android/src/androidTest/java/bbct/android/common/navigation/test/NavigateUpFromAboutTest.java

```diff
@@ -0,0 +1,144 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package bbct.android.common.navigation.test;
+
+import android.content.Context;
+import android.support.test.espresso.NoMatchingViewException;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import bbct.android.common.R;
+import bbct.android.common.test.DatabaseUtil;
+import bbct.android.lite.provider.LiteActivity;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+
+
+@RunWith(AndroidJUnit4.class)
+public class NavigateUpFromAboutTest {
+    @Rule
+    public ActivityTestRule<LiteActivity> activityActivityTestRule = new ActivityTestRule<LiteActivity>(LiteActivity.class);
+    Context context = null;
+    private DatabaseUtil dbUtil = null;
+
+    @Before
+    public void setUp() throws Exception {
+        activityActivityTestRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        context = getInstrumentation().getTargetContext();
+        this.dbUtil = new DatabaseUtil(context);
+    }
+
+    private void skipSurvey() {
+        try {
+            onView(withText(R.string.survey1)).check(matches(isDisplayed()));
+            onView(withText(R.string.later))
+                    .check(matches(isDisplayed()))
+                    .perform(click());
+        } catch (NoMatchingViewException e) {
+            //view not displayed logic
+        }
+    }
+
+    @Test
+    public void testNavigateToBaseballCardDetailsThenAboutFragment() {
+        skipSurvey();
+        String expectedTitle = context.getString(R.string.app_name);
+
+        // pop to BaseballCardList first if db is empty
+        if (this.dbUtil.isEmpty()) {
+            onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        }
+        // navigate to BaseballCardDetails
+        onView(allOf(withContentDescription(R.string.add_menu), isDisplayed())).perform(click());
+
+        // navigate to About
+        openActionBarOverflowOrOptionsMenu(context);
+        String aboutTitle = context.getString(R.string.about_title);
+        onView(allOf(withText(aboutTitle), isDisplayed())).perform(click());
```

Comment:
These three lines are a good candidate for a private helper method in this test class.